### PR TITLE
Create alias for starting the smart mirror from anywhere.

### PR DESCRIPTION
**Pull request by DannyLuu**

### CODE REVIEW

**Comment by DannyLuu** [comment]
For my dev env, I have the smart-mirror installed somewhere else.

File: scripts/pi-install.sh [hunk 1]
```diff
@@ -149,6 +149,13 @@ if ! grep -q '(smart-mirror)' ~/.config/lxsession/LXDE-pi/autostart; then
 @xset dpms 0 0 0' ~/.config/lxsession/LXDE-pi/autostart
 fi
 
+# Add start commands in the user's bashrc.
+echo "export MIRROR_HOME=\"~/smart-mirror\"" >> ~/.bashrc
+echo "alias update-mirror='cd \$MIRROR_HOME && git pull && npm install'" >> ~/.bashrc
```
Comment (justbill2020) [inline scripts/pi-install.sh]: Change `$MIRROR_HOME` to `~/smart-mirror`

File: scripts/pi-install.sh [hunk 2]
```diff
@@ -149,6 +149,13 @@ if ! grep -q '(smart-mirror)' ~/.config/lxsession/LXDE-pi/autostart; then
...
+echo "alias start-mirror='cd \$MIRROR_HOME && npm start'" >> ~/.bashrc
```
Comment (justbill2020) [inline scripts/pi-install.sh]: Change `$MIRROR_HOME` to `~/smart-mirror`

File: scripts/pi-install.sh [hunk 3]
```diff
@@ -149,6 +149,13 @@ if ! grep -q '(smart-mirror)' ~/.config/lxsession/LXDE-pi/autostart; then
...
+echo "alias dev-mirror='cd \$MIRROR_HOME && npm start dev'" >> ~/.bashrc
```
Comment (justbill2020) [inline scripts/pi-install.sh]: Change `$MIRROR_HOME` to `~/smart-mirror`

File: scripts/pi-install.sh [hunk 4]
```diff
@@ -149,6 +149,13 @@ if ! grep -q '(smart-mirror)' ~/.config/lxsession/LXDE-pi/autostart; then
...
+echo "alias ssh-start-mirror='cd \$MIRROR_HOME && DISPLAY=:0 npm start'" >> ~/.bashrc
```
Comment (justbill2020) [inline scripts/pi-install.sh]: Change `$MIRROR_HOME` to `~/smart-mirror`

File: scripts/pi-install.sh [hunk 5]
```diff
@@ -149,6 +149,13 @@ if ! grep -q '(smart-mirror)' ~/.config/lxsession/LXDE-pi/autostart; then
...
+echo "alias ssh-dev-mirror='cd \$MIRROR_HOME && DISPLAY=:0 npm start dev'" >> ~/.bashrc
```
Comment (justbill2020) [inline scripts/pi-install.sh]: Change `$MIRROR_HOME` to `~/smart-mirror`

File: scripts/pi-install.sh [hunk 6]
```diff
@@ -149,6 +149,13 @@ if ! grep -q '(smart-mirror)' ~/.config/lxsession/LXDE-pi/autostart; then
 @xset dpms 0 0 0' ~/.config/lxsession/LXDE-pi/autostart
 fi
 
+# Add start commands in the user's bashrc.
+echo "export MIRROR_HOME=\"~/smart-mirror\"" >> ~/.bashrc
```
Comment (justbill2020) [inline scripts/pi-install.sh]: Remove environmental variable

**Comment by evancohen** [comment]
I don't think having an environmental var to store the path of the mirror is a bad idea, just means we can make things more extensible.

As for having a bunch of different command alias' in the format of `*-mirror` I think those will be hard to remember. I think the best approach we can take here is to alias the word `mirror` as `npm *` to be executed in the smart mirror directory (defined by the environmental var), which would allow for the following:

| Command  | Action |
| ------------- | ------------- |
| `mirror start`  | Start the mirror  |
| `mirror start dev`  | Start the mirror with dev context  |
| `mirror run update` | Update the mirror |
| `mirror run sonus` | Run the sonus detector standalone |
| `mirror run train-model` | Run model training module |

... etc (I don't think the rest need to be captured here, but you get the idea)

To give us the capabilities described above we'd want to use a subshell function that can accept parameters, and then alias to that. This also gives us the added benifit of not changing the active shell directory.  The `.bashrc` should look like:
``` bash
export MIRROR_HOME=~/smart-mirror
run_mirror() { ( cd $MIRROR_HOME && DISPLAY=:0 npm "$@" ); }
alias mirror=run_mirror
``` 

(Be super careful with spaces here, bash is very picky)

**Comment by justbill2020** [comment]
@evancohen I agree way better approach

**Comment by evancohen** [comment]
Ooh, one more thing. In order to reload the bash profile in the install script we should call 
```
. .bashrc
```
from `~`

Otherwise the command wont work until reboot

File: scripts/pi-install.sh [hunk 7]
```diff
@@ -158,10 +163,10 @@ cat << "EOF"
       / _ \      For configuration instructions, check out the documentation at:
      |.o '.|     http://docs.smart-mirror.io
      |'._.'|     
-     |     |     To start your mirror you can run:
-   ,'|  |  |`.   > npm start
-  /  |  |  |  \  Or if you are running over SSH:
-  |,-'--|--'-.|  > DISPLAY=:0 npm start
+     |     |     To start your mirror you can run both on the pi or via SSH:
+   ,'|  |  |`.   > start-mirror
```
Comment (evancohen) [inline scripts/pi-install.sh]: `mirror start`

**Comment by DannyLuu** [comment]
@evancohen frig I thought I changed that aha.

**Pull request merged**